cpp/src/io/parquet/page_enc.cu

```diff
@@ -1966,9 +1966,9 @@ dremel_data get_dremel_data(column_view h_col,
 
     // Scan to get distance by which each offset value is shifted due to the insertion of empties
     auto scan_it = cudf::detail::make_counting_transform_iterator(
-      column_offsets[level], [off = lcv.offsets().data<size_type>()] __device__(auto i) -> int {
-        return off[i] == off[i + 1];
-      });
+      column_offsets[level],
```

Comment:
Why don't we make the `make_counting_transform_iterator` length be one less? It looks like the final output ( via thrust::scatter  ) already drops the last compute value. This would allow us to push up the length reduction and removal the conditional

Reply:
Because the `thrust::exclusive_scan` will get the wrong result.